### ISSUE TITLE
Block's given Receipt Hash doesn't match for block #1694983 (and some others)

### ISSUE THREAD

**Issue by Nashatyrev** [ethereum/ethereumj#431]
Block's given Receipt Hash doesn't match: `a192cae5e126dd5b549b30ecf6e614a16514a9ca23c58b600d1847637df4d150 != 4e24a12b44990332fffe133ba39d2107d6d549c31c0ab0a257bc8a0b000922eb`

Though stateRoot hash matches and we have no conflicts importing this block 

```
BlockData [ hash=cea35b399e548cdaa4bc309569a9601a1c02ae79ae65c78b23d307d452896f0a
  parentHash=dc8dd62bfa0b8f5887da0e9c1e6b2a7d90fa846f1106ed2d7b8ba8012946f785
  unclesHash=1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347
  coinbase=4bb96091ee9d802ed039c4d1a5f6216f90f81b01
  stateRoot=54c57cf3348ada52bd21e634feea158f80b141fab1fcd8e7bdd4487f02b9d75c
  txTrieHash=26d6a9dd811506dd6840002d55d490d3f6284d2bbe27f50718969e0231c5efae
  receiptsTrieHash=a192cae5e126dd5b549b30ecf6e614a16514a9ca23c58b600d1847637df4d150
  difficulty=2b45e9f6098f
  number=1694983
  gasLimit=47e7c4
  gasUsed=289100
  timestamp=1465798669 (2016.06.13 09:17:49)
  extraData=d783010404844765746887676f312e352e31856c696e7578
  mixHash=786926d4897b9d10ade6c0fc61259f43f5cfd1bc4e170dc2d92b25699561dcd9
  nonce=da45e958c6424fcd
Uncles []
Txs [
TransactionData [hash=a4a844e92c70d0c0009dd1c26611035f51871969de639e6329bfcadc17bc36ac  nonce=9083, gasPrice=04a817c800, gas=030d40, receiveAddress=18ccb7cffaaa42fc9344e4ed72e3c5ccc75fa978, value=, data=27dc297e4d0ab50e7bdc064d028a643a701ff1581aba27c9a70d4e5dc04412c3d9b6502e000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000063431363036360000000000000000000000000000000000000000000000000000, signatureV=28, signatureR=ef690a0ee104d94ecbddca2fc2f0a74b256e7abd39c3994c55b86b1dcb7dd51b, signatureS=5395fe334e1bfbe90d022132df2a9d5ed5af3f36e7ba127073d353435f20daf6]
TransactionData [hash=fe4b681beecb081fc61c0106c97a2b54ef81e6c07532982493c8a0c6827ea97b  nonce=9084, gasPrice=04a817c800, gas=030d40, receiveAddress=58d13795756bab016dee6afdc01e0fabb7f6fc70, value=, data=27dc297eb9e5ad5fc9a726958e468a62411139e49b46f12f7601b1d8b2a526d2dd68363a000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000063431363036360000000000000000000000000000000000000000000000000000, signatureV=28, signatureR=46410f395bfdedbe11e2836916f80220acd9ddd21804fb483f21a14623100577, signatureS=16eb5947055215e957fb7c25f1eec791b37122c9c7c2937a6e11ac2e9e6d30ee]
]
```

**Comment by Melvillian** [comment]
I just saw a similar error, but on block 1954654. Here's the log output

`java.lang.NullPointerException: null
        at org.ethereum.trie.TrieImpl.delete(TrieImpl.java:298)
        at org.ethereum.trie.TrieImpl.insertOrDelete(TrieImpl.java:215)
        at org.ethereum.trie.TrieImpl.update(TrieImpl.java:145)
        at org.ethereum.trie.SecureTrie.update(SecureTrie.java:25)
        at org.ethereum.trie.SecureTrie.delete(SecureTrie.java:30)
        at org.ethereum.db.ContractDetailsImpl.put(ContractDetailsImpl.java:75)
        at org.ethereum.db.ContractDetailsCacheImpl.commit(ContractDetailsCacheImpl.java:163)
        at org.ethereum.db.RepositoryTrack.commit(RepositoryTrack.java:339)
        at org.ethereum.core.BlockchainImpl.applyBlock(BlockchainImpl.java:830)
        at org.ethereum.core.BlockchainImpl.processBlock(BlockchainImpl.java:797)
        at org.ethereum.core.BlockchainImpl.add(BlockchainImpl.java:523)
        at org.ethereum.core.BlockchainImpl.tryToConnect(BlockchainImpl.java:405)
        at org.ethereum.sync.SyncManager.produceQueue(SyncManager.java:238)
        at org.ethereum.sync.SyncManager.access$200(SyncManager.java:35)
        at org.ethereum.sync.SyncManager$4$1.run(SyncManager.java:130)
        at java.lang.Thread.run(Thread.java:745)
07-27 01:56:54.626 [SyncQueueThread] ERROR sync - Block dump: f90bebf90211a029d61e84d6e01854bc8843134fdaa3387297e53b4d28abaece10f3df12ad0434a01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347944bb96091ee9d802ed039c4d1a5f6216f90f81b01a00b6aaa628c20bbc802273283433d1a23d6fbe792dde1c7285799f17eb83d1abaa079d21392d662072db665d496175693ab8f4f95b73a2858c7e5a04632aded5123a09b9d7277538b500e97f3901d2b4a167c8d5690bacb137e2d5a602b7d3a9755dcb90100000000000000000204000000000201000010000000000000000000000000000000080100000000000000000000004000000000000000000000000000000020000000000000000000000000080000000000000000000000004000000000080010000000000000100000400000000000000000000000000000000000100000000000000000000000000004000000000000000000000000900000010000000010000000000000000010000000000000000000000000000000000000000000000000000008020000000001000000000000000000000000420000000000001000000000000400000000000000000000000020000000000000000000000000000000008635f55b43abb3831dd35e8347e7c4830a4296845797150a91657468706f6f6c2e6f7267202855533129a0db32aa6b243f55a245a1d2ae4797e910eee21abdf205cf63ce099ed5313edb2188a2058eb00768ef57f909d3f86d822d1c8509502f90008252089428e61ecdf90732a7b0ce5720e6f1d157f1633a6a870f2601acb8741d801ba09689a75963fc1c4a96ef940f0a1148bcc4f8758528ac91e764e2e4bf80d63d88a056369a7fde712f7594589c5f5203f2fcb660db4c59fa2d8ee8d7c0ceea3af864f8708301cbe88506fc23ac0083051615947d4e28878325b51112946edbf5b65f126d222fad88024fe615355d3400801ba090394492f887d05dcbc8745d08716e5036deb2c42cc2a9018b55d1976bfde966a01823cc6e974fc092e8b6bd3aae02f25c59d7c78bc6b8c643143a8eed4a732193f8718301cbe98506fc23ac0083051615948062ed8b4537c4f58adb4fadc0fa927815130e5589022af9051fb7df0000801ba0e84315cb28aec5726e4207e97b591b9c1d48ef0142eaa2d4acc133891be448c1a03a14d93f4d3cc00ba43db01d480979dfa1df1ecf2fc8639f6449c45ba1e89076f869038504e3b2920083026cfb94bf4ed7b27f1d666546e30d74d50d173d20bca75480843ccfd60b1ba0ef7ac4482e95770632d12dec58056911a82d6daca7edf7e9c4fd1952f4bf09b9a03e1b00f883ff9bc5721facf3ff061e242e212fd5995fb1ef54ed3a54564e516ef86d808504e3b29200825208945924fa3b7c1b8b739e932939c631881ee4faeaf489056bc75e2d63100000801ba0f52d565444a6aecc9875129f900a5023a23bca92486d7e580849e2978f73d992a076c9cde6eb6baf125ba5ab7b043a72cb1fa5c2c2b6c92818bd5e31790e54e24af870830505848504a817c80083015f9094c1c427cd8e6b7ee3b5f30c2e1d3f3c5536ec16f5880e0618e3ee719c84801ca07c5c35b865532fccd79fc393be0b1dee474c8aa06cf1d35249b0dac8d5dc6669a03c27fbc7557191e648c62ecab3246e4af9616c158e9cb903db2479e3c6adf81bf86c668504a817c8008252089491337a300e0361bddb2e377dd4e88ccb7796663d880b4cefcdd6e6c800801ca08edfd79278d9f29338bbd93fa5ba07ee7994d1b815efbce0c199016c7c6bc392a03325974a3869532b5c922818edd8f3af62daf64352375ee0c8e53ef1aa1793e3f86f830505858504a817c80083015f9094c2654abc704e589d826fe3f69f4ab21fb5660a9a8806e27f2341fda008801ba077a13dc2b2f39f3d5511cf336a950dbec63f7b153d8c44994ec60cb8effbc2539fd62d46ef68b49d06ee0f24de0f9d33b3fa96fcf0f8cb6b941fdb601ad83f52f86e8223538504a817c800830f424094b09960d5f64e5709b305aaa2de8ecd85a2626905872386f26fc10000801ca0b0189e3ca558802e46dc6a3248e254476b65107182c5516fffccdf5b23bf570da05681a2c2a7e0a4814eea5e16969b62cfbe0208d17ad1c9ce9cab9890690bb646f870830505868504a817c80083015f9094cf08121b87dd99fce95156913112733b462afc15880de769c5bb84f71c801ca02f2fd9cdc554bc5e7787c6d4fd103ab6cc367ce2e82a0d68738daf93afbb7384a0485e624c7850d9f410b2e3163d142b7116e2305a8a4af54ba6cc66c9d32f5c6ef8ac8223548504a817c800830f424094fdc77b9cb732eb8c896b152e28294521f5f62e6780b844e637a44b0000000000000000000000000000000000000000000000000002fcec8ee3a9e00000000000000000000000000000000000000000000000000002e70d1d356a3f1ca00440e89f8c8e35919afa1eae4a254f7ae85536eafbc19924e4b13b44e14f1a8ca05abc75da3771e79fd0e1b998b5585aef9ab0afa760ad6b8f961016bffd39ddeef870830505878504a817c80083015f9094d0ba080f37afda08f10f2769770ec27498c299ae880de3f006b9247628801ba0391b45734077f36a38a3aeaa2ec1a9b2bdbe5723afdbb6a6891fb1b47de73d1ca074b8aec621616034beaae5d19f0d0c7a6c22a064eb63dcd0c6c7c8db64e1bd1ff86d018504a817c80082520894dec23f41d7dfaf36518e055e62b215ad8cb755d1890f3bc41ce088390000801ca08b7f9d0ef298937d10591e12f4a9c0ba579d5c2582dcd697855bb55a489b3bffa031e6d9455a2d97e5dca5f4cada0d033047ecc005f6d5fa363d24a3afc3304809f870830505888504a817c80083015f9094d46787146b1bd34cedf89f54144717a7089dc6eb880e033c41c5c576b0801ca03e3f7d4c9517429efc09004a8b52239486640b7e3394b2e7b3f66e25975dcc22a01a75517bfb64c36095dfaf9000350f8cf428de430f348a61f4f198c1f8490d43f9014c823d9b8504a817c80083124f809441f274c0023f83391de4e0733c609df5a124c3d480b8e490fa337d000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000038d7ea4c6800000000000000000000000000001a23fd30aabf5ad53bab3093dcd4948e15cec80000000000000000000000000000000000000000000000000000000000000005000000020df8892e9adae42832690036036eda1c586c3299625036f000000000000000000c7ac55fc4627bb62958317beb1082c1f04be8901d1cc3ab5e5307bd17482c764f013975769260518659f5ffd000000000000000000000000000000001ca09cf9be25aca7f198628d3acf726ad6386a900da700fe3e2a86716492df27739da06208aa59ccb4e75e0c6f25e2f615d9ab15e16fa2315adc2f6eaf6458960594c6f86c0e8504a817c80082c3509458ae42a38d6b33a1e31492b60465fa80da595755880de19c951e4862d4801ca0648fef100b5f89ee027a44d27e5e8f5f7a032b86b41f576f2fce8c87337c1136a049383ce552eadd13f3ff2f24d8e44dc68edd5d90de4218e0bb196279d2afe22bf86b068504a817c8008261a894804c86c2822364a5c050749323f60ed2b3941f2a872386f26fc10000801ba0d857054a66b127e2d23e220cb6bb3a410ab18fa5981de1e436ed0c73bc13158ea02a2aca3830c3860d9f6c41d2c2d881a0348fa92da0c054c5429e6efa63b34900f86b078504a817c8008261a894804c86c2822364a5c050749323f60ed2b3941f2a872386f26fc10000801ba0098ef1376fb7f5cfb295d6e52c131bf898df9722872f545afb408ad89f8f50e4a02c5cb4f23ac8dab0d8ff5d525cea28e712e8e146e88cb724b7899eac099261acf86b088504a817c8008261a894804c86c2822364a5c050749323f60ed2b3941f2a872386f26fc10000801ca0f1492f6b98049aac46957c8718b3ea6b54f2468fa3a7e5a4b10bf3434d38bb7aa042d41c6b27f9063f2d1b9100febd606eb88e4c95123dcef3724bd312caeb5a05f86b098504a817c8008261a894804c86c2822364a5c050749323f60ed2b3941f2a872386f26fc10000801ba0e0e4567ea2cce8812271be88856a1cd764d90b06dd6c0ef892bd2a300a8b58cfa06f89b3e0d23672eccc348bb110c84a3432ddaade6060fc09203fde9ff0881f07c0`

**Comment by Nashatyrev** [comment]
@Melvillian It's a different issue. Is it reproducible when you restart without DB clean up?

**Comment by Melvillian** [comment]
@Nashatyrev How could I reproduce it without performing a DB cleanup? ethJ has already processed that block, so I can't undue it and replay it without wiping the database directory and starting over. I did see this same error again, though this time it was block #1513959 

```
07-28 21:45:33.837 [SyncQueueThread] ERROR sync - Error processing block #1513959 (e1c047 <~ c06d3f) Txs:9, Unc: 0:
java.lang.NullPointerException: null
        at org.ethereum.trie.TrieImpl.delete(TrieImpl.java:298)
        at org.ethereum.trie.TrieImpl.insertOrDelete(TrieImpl.java:215)
        at org.ethereum.trie.TrieImpl.update(TrieImpl.java:145)
        at org.ethereum.trie.SecureTrie.update(SecureTrie.java:25)
        at org.ethereum.trie.SecureTrie.delete(SecureTrie.java:30)
        at org.ethereum.db.ContractDetailsImpl.put(ContractDetailsImpl.java:75)
        at org.ethereum.db.ContractDetailsCacheImpl.commit(ContractDetailsCacheImpl.java:163)
        at org.ethereum.db.RepositoryTrack.commit(RepositoryTrack.java:339)
        at org.ethereum.core.BlockchainImpl.applyBlock(BlockchainImpl.java:830)
        at org.ethereum.core.BlockchainImpl.processBlock(BlockchainImpl.java:797)
        at org.ethereum.core.BlockchainImpl.add(BlockchainImpl.java:523)
        at org.ethereum.core.BlockchainImpl.tryToConnect(BlockchainImpl.java:405)
        at org.ethereum.sync.SyncManager.produceQueue(SyncManager.java:238)
        at org.ethereum.sync.SyncManager.access$200(SyncManager.java:35)
        at org.ethereum.sync.SyncManager$4$1.run(SyncManager.java:130)
        at java.lang.Thread.run(Thread.java:745)
07-28 21:45:33.837 [SyncQueueThread] ERROR sync - Block dump: f90a3af9021aa0c06d3f9bcc521603235ccd1b60f0aee29be0a3dffc7e30c6140af66396535e2aa01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d4934794f8b483dba2c3b7176a3da549ad41a48bb3121069a05d4d559552f456274a6f05b1cb357276e3e8f2f0c9974883be10ff47b3fcc033a0ac8fb21ae81cbc6a9b1a9b0a1cfc73f7b723d1a92ff77d9e6c63c7aa3135c509a0aa26a2d595f73dbe448b199937ed02c52ddc730184adc199d0f73d11ba9764a7b9010000300000000000020000000004020000000000010000000000000000000000000001000010000000000000000001000000000000000000400000000000000000000000000000000000000008000000000000000000000100000100000000000000400000020000000000000000000b020000000000000000000000100000000000000000000000001100000000000000000000000000000880000000000010000000000000000000000000000080000000000000000004080000000020000100000000020000000000000400002000000000000000000200000010000000200000000000000000200000000000020000000000000000000000000000200000008621b4f117e4e4831719e78347e7c48306f1ef845736b3939ad983010305844765746887676f312e342e328777696e646f7773a09f68ffaaa8188df0cdb6db9ff0b0af4c8ac10ab1aae9c046727ce8c47e2c43fb88fc00e6f1987b868ef90819f901cd82642d8504a817c8008307a12094ad62f56a03334b647e55dbdb5b8642c24605a80101b901643ed486790000000000000000000000002e6480c43704100a7f60f6006d94aff5a5801f49000000000000000000000000000000000000000000000000000000005736b39000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000bd2b934a7e30c0753c6d9e8774e1f90b56bd23ad00000000000000000000000000000000000000000000000000000000000f1b30000000000000000000000000000000000000000000000000000000000000000257c8abf2b07b2bb24e582c586486414961ab1eaeb5f8d0a308e7bb38cb8763ef5e9b72cc3888209cca843d09c7b357b01ee0a6dd9131ee0cea6d3cda08f24d7f0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000001b1ba06554dc599f5fb02bc974a55fb0a164e6aa249beb36150be4f17ad42aed2fd3d4a07671c20c16f4ad245aca74259b6d080ee00819723b0daa0dea00f4f192fde7a4f870826c6b8504a817c800830249f094ad313e7425f2e1f28a011b37326b48c135892c95890ad77cf8e18e3f8000801ca06834f1b2619754b595160d0bfdf627b6b7c650eaf25132ea30ce07681e4e34b4a016b8b30436081b0028c85a0e59be5e5f15874a003f74e9ac8b5dabc7c23c2d17f901cd82642e8504a817c8008307a12094ad62f56a03334b647e55dbdb5b8642c24605a80101b901643ed486790000000000000000000000002e6480c43704100a7f60f6006d94aff5a5801f49000000000000000000000000000000000000000000000000000000005736b39200000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000001200000000000000000000000009b58e1ac6b2cc641f452d10e4e566060dcbbeea000000000000000000000000000000000000000000000000000000000014c562000000000000000000000000000000000000000000000000000000000000000022936c37bbbcf7a3c203e3a2408a36fc708693762a89ad9c0bb296b764345ce4472aeac08767e5fa384fdc36c3963c1b39b0d7b4f218ae1a175a41f06370247790000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000001c1ba0eb80a19364285337cf7b275a1aea242c611ecc961ce9d2f6931f458968087fe8a043c2324f8bbb8185e5ead3936942e186c5daabf553762966ba55e2de6b262dfcf86f826c6c8504a817c800830249f094ddf73b0384f5fc6ce09ea50dd08d698c6612ce1688015181ff25a98000801ca0724c0fb81e934f0906029acd52531b2714c29bb5a68ad23319d2e1de02ba6e96a009fb5988421f112ee10c070b49625f95cbbdc4e7c243ee3765b6227a9e4717d0f901cd82642f8504a817c8008307a12094ad62f56a03334b647e55dbdb5b8642c24605a80101b901643ed486790000000000000000000000002e6480c43704100a7f60f6006d94aff5a5801f49000000000000000000000000000000000000000000000000000000005736b39300000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000001200000000000000000000000003d278d0c6aa634df61e8908edafc23b43260533f0000000000000000000000000000000000000000000000000000000000e297d000000000000000000000000000000000000000000000000000000000000000021d90c94fcc81a2a222e150dff61a33e58eb425b24e1cff2c2899ac25287cfd430eedd1e6af803c9feed413a7f575bcfee5a14781a4ba00e6cdc949d428d2b8f30000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000001c1ca0cb9237387856ae8b5ae3f89122182de27bc2c5e525e6417d6aba3f6e69b1686fa03bee3d9aeda3a907a8c1d252c95812bb7a55e710a230ce53c220c8417797533af870830960f28504a817c80083015f90945fc70202e392cf3a97c9cd0d82e4e36dbe738079880fd46ed5c263fc00801ba0d6731e1230754f6d61069083def741f8352bfb0edfb1ff6ae8dab04b7657438da01cbebf5149b75fa8d04041966555053b3f9065226b17a4008319b3828090af94f870830960f38504a817c80083015f9094fc8bc4ed86df73001ce6082f3db914d7c7a9a6cc880fcc711722cdec00801ba0f1bf5572caeb5d7b18394b336a5608a31a2b137d386ec7f43af46f14cee63267a05188f7a6deb09694d18028e71ae4a84bc34076ac7b3247f2c750ceda29a8d7acf870830960f48504a817c80083015f9094b172ee078e9a51beb22be3e3ee281a0bdee117c9880fc9ae2adc8f5000801ba0bb976fab44116418419d10797b2ef3d52b8b155cb27576a2a82a370ffefb9cd0a0117b82be1f1fbe3195d9a438f280937a5c3069158cbd54873277da9376687b66f86e8085043bbb4d808302a80b94bb9bc244d798123fde783fcc1c72d3bb8c18941389394c20e6837d5c2000801ca017b57501662d630e1f81d02f18a667671487d8059018f7d089164ec5f913aeb6a0095374c1b89facb5622f55694bae368b63f880275daeec1d7ae794ab7e99fc4bc0
```

**Comment by zilm13** [comment]
Fixed in #523
